lib/galaxy/tools/__init__.py

```diff
@@ -837,10 +840,20 @@ def __parse_legacy_features(self, tool_source):
                     self.hook_map[key] = value
             file_name = code_elem.get("file")
             code_path = os.path.join(self.tool_dir, file_name)
-            with open(code_path) as f:
-                compiled_code = compile(f.read(), code_path, 'exec')
             if self._allow_code_files:
-                exec(compiled_code, self.code_namespace)
+                with open(code_path) as f:
+                    code_string = f.read()
+                try:
+                    compiled_code = compile(code_string, code_path, 'exec')
+                    exec(compiled_code, self.code_namespace)
+                except Exception:
+                    if refactoring_tool and self.python_template_version.release[0] < 3:
```

Comment:
Why `if refactoring_tool` ?
```suggestion
                    if self.python_template_version.release[0] < 3:
```